src/dataconnect/schemaMigration.ts

```diff
@@ -53,51 +54,74 @@ export async function migrateSchema(
   if (!instanceId) {
     throw new FirebaseError(`tried to migrate schema but ${instanceId} was undefined`);
   }
-  const iamUser = await setupIAMUser(instanceId, databaseId, options);
   try {
-    // TODO(b/330596914): Handle cases where error only comes back after validateOnly=false
-    await upsertSchema(schema, /** validateOnly=*/ true);
+    await upsertSchema(schema, validateOnly);
```

Comment:
```
if validateOnly && existingSchema.schemaValidation == "NONE" {
   // Call upsertSchema(existingSchemaWithSchemaValidationStrict, validateOnly=false)
   // This ensure the backend connectivity to the Cloud SQL instance is provisioned.
   // Without it, `upsertSchema(schema, validateOnly=true)` below may success with false positives
}
```

Reply:
Can you make sure to test `dataconnect:sql:diff` on a fresh new instance created by Firebase Console?

Reply:
Just tested, and it reports no differences. However, I'm not sure how to safely fix this - we can't call `upsertSchema(schema, validateOnly=true)` from sql:diff - if it succeeds, it would update the schema instead of just showing us the difference

Reply:
Thanks for the offline chat - implemented your suggestion and verified that both diff and migrate work on the 'fresh  outta console' state.